src/vs/workbench/contrib/inlineChat/browser/inlineChatDecorations.ts

```diff
@@ -0,0 +1,161 @@
+	private inlineChatLineNumber: number | undefined;
+	private gutterDecorationID: string | undefined;
+	private gutterDecorationLine: number | undefined;
+
+	public static readonly gutterSettingID = 'inlineChat.showGutterIcon';
+	private static readonly gutterIconClassName = 'codicon-inline-chat';
+
+	private static readonly GUTTER_DECORATION = ModelDecorationOptions.register({
+		description: 'inline-chat-decoration',
+		glyphMarginClassName: ThemeIcon.asClassName(gutterInlineChatIcon),
+		glyphMargin: { position: GlyphMarginLane.Left },
+		stickiness: TrackedRangeStickiness.NeverGrowsWhenTypingAtEdges,
+	});
+
+	constructor(
+		private readonly editor: ICodeEditor,
+		@IInlineChatService private readonly inlineChatService: IInlineChatService,
+		@IConfigurationService private readonly configurationService: IConfigurationService,
+		@IContextKeyService private readonly contextKeyService: IContextKeyService
+	) {
+		super();
+		this._register(this.configurationService.onDidChangeConfiguration((e: IConfigurationChangeEvent) => {
+			if (!e.affectsConfiguration(InlineChatDecorationsContribution.gutterSettingID)) {
+				return;
+			}
+			this.onEnablementOrModelChanged();
+		}));
+		this._register(this.inlineChatService.onDidChangeProviders(() => this.onEnablementOrModelChanged()));
+		this._register(this.editor.onDidChangeModel(() => this.onEnablementOrModelChanged()));
+	}
+
+	private onEnablementOrModelChanged(): void {
+		// cancels the scheduler, removes editor listeners / removes decoration
+		this.localToDispose.clear();
+		const model = this.editor.getModel();
+		if (!model || !this.isSettingEnabled() || !this.hasProvider()) {
+			return;
+		}
+		const decorationUpdateScheduler = new RunOnceScheduler(() => this.onSelectionOrContentChanged(model), 200);
+		this.localToDispose.add(decorationUpdateScheduler);
+		this.localToDispose.add(this.editor.onDidChangeCursorSelection(() => decorationUpdateScheduler.schedule()));
+		this.localToDispose.add(this.editor.onDidChangeModelContent(() => decorationUpdateScheduler.schedule()));
+		this.localToDispose.add(this.editor.onMouseDown(async (e: IEditorMouseEvent) => {
+			if (!e.target.element?.classList.contains(InlineChatDecorationsContribution.gutterIconClassName) || !this.editor.hasModel()) {
+				return;
+			}
+			const startLineNumber = this.editor.getSelection().startLineNumber;
+			const inlineChatPositionUnchanged = startLineNumber === this.inlineChatLineNumber;
+			const inlineChatVisible = this.contextKeyService.getContextKeyValue<boolean>(CTX_INLINE_CHAT_VISIBLE.key);
+			if (inlineChatPositionUnchanged && inlineChatVisible) {
+				return;
+			}
+			this.inlineChatLineNumber = startLineNumber;
+			InlineChatController.get(this.editor)?.run();
+		}));
+		this.localToDispose.add({ dispose: () => this.clearDecorations() });
+		decorationUpdateScheduler.schedule();
+	}
+
+	private onSelectionOrContentChanged(model: ITextModel): void {
+		const selection = this.editor.getSelection();
```

Comment:
To avoid this we could pass `editor: IActiveCodeEditor` as argument.

For that to get typed we need to change the code in `onEnablementOrModelChanged`:

		if (!this.editor.hasModel() || !this.isSettingEnabled() || !this.hasProvider()) {
			return;
		}
		// now this editor is IActiveCodeEditor
		const model = this.editor.getModel();

Reply:
Since onSelectionOrContentChanged is inside of a run once scheduler the compiler warns that `this.editor.getModel()` can be null when passed in as an argument. The check should be inside of the method.